lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -114,12 +119,15 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
         });
     }
     const backendInfo = backendInfoObj.backendInfo;
+    const location = backendInfo.getControllingLocationConstraint();
+    const locationType = config.locationConstraints[location].type;
+    const skipBackend = ['aws_s3'];
 
     const requestLogger =
         logger.newRequestLoggerFromSerializedUids(log.getSerializedUids());
     return async.waterfall([
         function storeData(next) {
-            if (size === 0) {
+            if (size === 0 && skipBackend.indexOf(locationType) < 0) {
```

Comment:
Instead of going through an array, it might be faster to create an object 
**ex**: `const skipBackend = { 'aws_s3': true };` and check for `skipBackend[locationType]`